### ISSUE TITLE
Bug with Style/HashSyntax shorthand autocorrection and ruby19_no_mixed_keys

### ISSUE THREAD

**Issue by tycooon** [rubocop/rubocop#11409]
## Steps to reproduce the problem

1. Set `EnforcedStyle: ruby19_no_mixed_keys` for `Style/HashSyntax` cop.
2. Write the following ruby code:

```ruby
bar = 42
foo("some-value" => true, bar:)
```

## Actual behavior

`rubocop -a` renders invalid ruby code that looks like this:

```ruby
bar = 42
foo("some-value" => true, :bar => )
```

The rubocop itself approves that:
```
test.rb:5:3: E: Lint/Syntax: unexpected token tRPAREN
(Using Ruby 3.2 parser; configure using TargetRubyVersion parameter, under AllCops)
```

## RuboCop version

```
1.42.0 (using Parser 3.2.0.0, rubocop-ast 1.24.1, running on ruby 3.2.0) [arm64-darwin22]
```
